src/helpers/ht_device_info.service.ts

```diff
@@ -70,26 +50,344 @@ class DeviceInfoService {
+            return this.getUAFromNavigator().os;
+        }
+
+        return null;
+    }
+
+    getBrowser() {
+        if ( this.browserData && this.browserData.browser) {
+            return this.browserData.browser.name;
+        } else {
+            return this.getUAFromNavigator().browserName;
+        }
+
+        return null;
+    }
+
+    getBrowserVersion() {
+        if ( this.browserData && this.browserData.browser) {
+            return this.browserData.browser.version;
+        } else {
+            return this.getUAFromNavigator().fullVersion;
+        }
+
+        return null;
+    }
+
+    getBrowserMajorVersion() {
+        if ( this.browserData && this.browserData.browser) {
+            return this.browserData.browser.version;
+        } else {
+            return this.getUAFromNavigator().majorVersion;
+        }
+
+        return null;
+    }
+
+    getFonts() {
+        return this.fonts;
+    }
+
+    getFingerprint() {
+        if ( !this.fingerprint ) {
+            if ( this.fingerprintComponents && Object.prototype.toString.call( this.fingerprintComponents ) === '[object Array]') {
+                const values = this.fingerprintComponents.map(function (component) { return component.value });
+                this.fingerprint = Fingerprint2.x64hash128(values.join(''), 31);
+                this.pushFPC(this.fingerprintDetails, 'fingerprint', this.fingerprint);
+            } else {
+                return null;
+            }
+        }
+        return this.fingerprint;
+    }
+
+    getFingerprintComponents() {
+
+        if ( !this.fingerprintDetails ) {
+            this.fingerprintDetails = new Array();
+
+            if ( this.fingerprintComponents ) {
+                for ( let i = 0; i < this.fingerprintComponents.length; i++ ) {
+                    if ( this.fingerprintComponents[i] &&
+                        this.fingerprintComponents[i].key !== 'canvas' &&
+                        this.fingerprintComponents[i].key !== 'webgl' ) {
+
+                        if ( this.fingerprintComponents[i].key === 'js_fonts' ) {
+                            this.fonts = this.fingerprintComponents[i].key;
+                        }
+                        this.fingerprintDetails.push({
+                            key: this.fingerprintComponents[i].key,
+                            value: this.joinValues(this.fingerprintComponents[i].value)
+                        });
+                    }
+                }
+
+
+                // FingerprintJS does not generate these
+                this.pushFPC(this.fingerprintDetails, 'osType', this.getOsType());
+                this.pushFPC(this.fingerprintDetails, 'browserType', this.getBrowser());
+                this.pushFPC(this.fingerprintDetails, 'browserVersion', this.getBrowserVersion());
+                this.pushFPC(this.fingerprintDetails, 'deviceName', this.getDeviceName());
+                this.pushFPC(this.fingerprintDetails, 'userAgent', navigator.userAgent);
```

Comment:
Minor: No need an action now. I have tested with Console, and it seems that the userAgent is already generated from the fingerprintJs, so we could delete the line 147 related to "userAgent"

Reply:
The this.pushFPC call only actions on entries that don't exist, and its just in case FingerprintJS doesn't do this on certain browsers.

Added to https://www.pivotaltracker.com/story/show/171322089